Build/Build.cs

```diff
@@ -354,7 +354,20 @@ void ReportTestOutcome(params string[] globFilters)
         .ProceedAfterFailure()
         .Executes(() =>
         {
-            Node($"{YarnCli} --silent install", workingDirectory: RootDirectory);
+            Node($"{YarnCli} --silent install", workingDirectory: RootDirectory,
+                logger: (_, msg) =>
+                {
+                    // See this PR here: https://github.com/fluentassertions/fluentassertions/pull/2537
+                    // and several comments and references.
+                    // This seems to be a bug in yarn and the package 'Yarn.MSBuild' is no longer maintained;
+                    // So: ignore 'yarn install' errors when the same cache directory is defined in yarn.lock.
+                    // This errors have the pattern like: 'Error: warning Pattern [packagename@version]...'
+
+                    if (!msg.Contains("warning Pattern"))
+                    {
+                        Error(msg);
+                    }
```

Comment:
I assume `warning Pattern` covers all kinds of warnings.
So how about we only completely ignore `"is trying to unpack in the same destination"` to avoid silencing too much?
```cs
if (!msg.Contains("is trying to unpack in the same destination"))
{
	if (msg.Contains("warning Pattern"))
	{
		Warning(msg);
	}
	else
	{
		Error(msg);
	}
}
```

Reply:
Is this really the case? If I googling "warning pattern", it only comes up with this sort of error.

Reply:
`Pattern` comes from
https://github.com/yarnpkg/yarn/blob/158d96dce95313d9a00218302631cd263877d164/src/reporters/lang/en.js#L127-L128

`Warning` comes from `config.reporter.warn(`
https://github.com/yarnpkg/yarn/blob/master/src/package-fetcher.js#L118-L123

grepping for `reporter.warn(` in the yarn src gives 60 hits.

Reply:
Hmm true, but the casing (with pattern in uppercase) is only used for this error.

But I will change it according to you suggestion :)